### PR TITLE
added rotateAngle class

### PR DESCRIPTION
**Pull request by Khalil-Burns**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/SwerveSubsystem.java [hunk 1]
```diff
@@ -151,6 +151,21 @@ public void drive(double xSpeed, double ySpeed, double rot, boolean fieldRelativ
...
 
+    public double getPitchValue() {
+        double[] ypr = new double[3];
+        m_pigeon.getYawPitchRoll(ypr);
+        return Math.abs(ypr[1]);
+    }
+    public double getPitch() {
+        return(m_pigeon.getPitch());
+    }
+    public double getYaw() {
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/subsystems/SwerveSubsystem.java:162]: remove this. Use getHeading()

File: src/main/java/frc/robot/subsystems/SwerveSubsystem.java [hunk 2]
```diff
@@ -151,6 +151,21 @@ public void drive(double xSpeed, double ySpeed, double rot, boolean fieldRelativ
...
+        return(m_pigeon.getYaw());
+    }
+    public void resetPigeon() {
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/subsystems/SwerveSubsystem.java:165]: remove this method as well

File: src/main/java/frc/robot/commands/RotateAngle.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    this.angle = _angle;
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    SwerveSubsystem.getInstance().resetPigeon();
+
+    this.newYaw = this.angle + SwerveSubsystem.getInstance().getYaw();
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/commands/RotateAngle.java:33]: remove getYaw()
Use SwerveSubsystem.getInstance().getHeading().getDegrees();

File: src/main/java/frc/robot/commands/RotateAngle.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    // Use addRequirements() here to declare subsystem dependencies.
+    addRequirements(SwerveSubsystem.getInstance());
+
+    this.angle = _angle;
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    SwerveSubsystem.getInstance().resetPigeon();
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/commands/RotateAngle.java:31]: remove this resetPigeon();

File: src/main/java/frc/robot/commands/RotateAngle.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {}
+
+  // Returns true when the command should end.
+  @Override
+  public boolean isFinished() {
+    if (Math.abs(SwerveSubsystem.getInstance().getYaw() - this.newYaw) <= tolerance) {
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/commands/RotateAngle.java:55]: try to use pidcontroller:  if( pidControl.atSetpoint())
    {
      return true;
    }
    else{
      return false;
    }

File: src/main/java/frc/robot/commands/RotateAngle.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    addRequirements(SwerveSubsystem.getInstance());
+
+    this.angle = _angle;
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    SwerveSubsystem.getInstance().resetPigeon();
+
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/commands/RotateAngle.java:32]: you can add these two lines in initialize()
pidControl.setTolerance(0.1);
    pidControl.enableContinuousInput(-180.0, 180.0);

**Review by ErikCald — DISMISSED** [review]

**Review by weizhang-2706 — APPROVED** [review]

**Review by weizhang-2706 — APPROVED** [review]

**Review by ErikCald — APPROVED** [review]

**Pull request merged**